ci/long_running_tests/workloads/serve_handle.py

```diff
@@ -0,0 +1,59 @@
+# A test that stresses the serve handles. We spin up a backend with a bunch
+# of replicas that just forward requests to another backend.
+
+import ray
+from ray import serve
+from ray.serve import BackendConfig
+import logging
+
+num_replicas = 3
+num_queries = 10
+
+ray.init(address="auto")
+client = serve.start()
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def hello_world(_):
+    return "Hello World"
+
+
+class forwardActor:
+    def __init__(self):
+        self.handle = client.get_handle("hello_world")
+
+    def __call__(self, _):
+        return ray.get(self.handle.remote())
+
+
+# Here the handle is created each time the function is called, so it may stress
+# the system more.
+def forward(_):
+    handle = client.get_handle("hello_world")
+    return ray.get(handle.remote())
+
+
+config = BackendConfig(num_replicas=num_replicas)
+
+client.create_backend("forward", forward, config=config)
+client.create_endpoint("forward", backend="forward")
+
+client.create_backend("hello_world", hello_world)
+client.create_endpoint("hello_world", backend="hello_world")
+
+client.create_backend("forwardActor", forwardActor, config=config)
+client.create_endpoint("forwardActor", backend="forwardActor")
+
+handle = client.get_handle("forwardActor")
+logger.info("Starting serve handle stress testing (actor)")
+for _ in range(num_queries):
+    ray.get(handle.remote())
+logger.info("Finished serve handle stress testing (actor)")
+
+handle = client.get_handle("forward")
+logger.info("Starting serve handle stress testing (function)")
+for _ in range(num_queries):
+    ray.get(handle.remote())
+logger.info("Finished serve handle stress testing (function)")
```

Comment:
I'm not sure stress test handle with the function tells us anything interesting. How about we just use the actor but varies the number of `hello_world` replicas? So the output tells us something like this:
```
With 1 forward replicas, 1 worker replicas, 1000 queries: 5.2s
With 1 forward replicas, 5 worker replicas, 1000 queries: 1.2s
With 1 forward replicas, 10 worker replicas, 1000 queries: 1.2s
Just calling the worker replicas, 1 worker replicas: 1000 queries: 3.2s
Just calling the worker replicas, 10 worker replicas: 1000 queries: 1.2s
```
*Note*: can we make the ForwardActor async?

What do we see here? 
- By comparing the forward replicas and just calling the worker replicas, we measure the (latency) overhead in handle. This answers the question: How much latency/throughput hit is there when I compose models?
- By comparing the total completion time as we fix the number of forward replica and varies the number of worker replicas, we measure the limit of a single async actor. This answers the question: How many "ForwardActor" or this kind of high level pipeline workers do I need to provision for my workload? every 1k qps, 2k qps?

lmk if this make sense or not. Happy to explain more.

Reply:
@simon-mo's suggestion sounds good to me! Can we also please report the results as queries per second instead of time to completion?